src/foam/core/stdlib.js

```diff
@@ -372,9 +372,8 @@ foam.LIB({
     function clone(o) { return o; },
     function equals(a, b) { return foam.Number.compare(a, b) == 0; },
     function compare(a, b) {
-      if ( ! foam.Number.isInstance(b) && isNaN(parseFloat(b))
-          || ( isNaN(a) && ! isNaN(b)) ) return 1;
-      else if ( ! isNaN(a) && isNaN(b) ) return -1;
+      if ( ! foam.Number.isInstance(b) || ( isNaN(a) && ! isNaN(b)) ) return 1;
+      if ( ! isNaN(a) && isNaN(b) ) return -1;
```

Comment:
Are the guards in the comparators still required at this point? Do we expect any users to directly call type compare with different types?